### PR TITLE
BUGFIX: SD-538 align axis to base axis

### PR DESCRIPTION
**Pull request by naibinh**
[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)
- [ ] Squash commits

# Related PRs
- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2355
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2126

# Related Jira tasks
- SD-538: https://jira.intgdc.com/browse/SD-538

### CODE REVIEW

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1680/ : FAILURE in 3m 43s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5460/ : SUCCESS in 9s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4280/ : FAILURE in 4m 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5459/ : SUCCESS in 8s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4282/ : SUCCESS in 12m 37s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5468/ : SUCCESS in 18s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2307/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1681/ : SUCCESS in 13m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5469/ : SUCCESS in 15s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4294/ : SUCCESS in 19m 49s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5485/ : SUCCESS in 12s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4298/ : FAILURE in 4m 00s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5487/ : SUCCESS in 13s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2317/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1691/ : SUCCESS in 13m 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5488/ : SUCCESS in 14s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2319/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1692/ : UNSTABLE in 12m 52s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5490/ : SUCCESS in 26s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4300/ : SUCCESS in 14m 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5489/ : SUCCESS in 20s

**Comment by naibinh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1079.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1693/ : SUCCESS in 28m 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5491/ : SUCCESS in 13s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4309/ : SUCCESS in 6m 02s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5497/ : SUCCESS in 7s

File: stories/internal/DualAxesChartConfig.tsx [hunk 1]
```diff
@@ -259,4 +259,31 @@ storiesOf("Internal/DualAxesMinMaxConfig", module)
     })
     .add("Column should not be cut off on left axis", () => {
         return screenshotWrap(<div>{renderSupportedCharts(dataSet.sd160DataSet02)}</div>);
+    })
+    .add("Chart should be zero aligned and no columns is hidden", () => {
+        // SD-538
+        const chartConfig: IChartConfig = {
+            type: "column",
```
Comment (ptquang86) [inline stories/internal/DualAxesChartConfig.tsx]: Please change to existing constant

Comment (naibinh) [reply stories/internal/DualAxesChartConfig.tsx]: done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4313/ : SUCCESS in 4m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5500/ : SUCCESS in 10s

**Comment by ptquang86** [comment]
LGTM

File: src/components/visualizations/chart/highcharts/plugins/adjustTickAmount.ts [hunk 2]
```diff
@@ -125,32 +125,56 @@ export function adjustTicks(axis: IHighchartsAxisExtend): void {
...
+    }
+    return score;
+}
+
+/**
+ * Base on axis score which is bigger than another, will become base axis
+ * The other axis will be aligned to base axis
+ * @param axes
+ */
+function getBaseAxis(axes: IHighchartsAxisExtend[]): IHighchartsAxisExtend[] {
```
Comment (lanhq) [inline src/components/visualizations/chart/highcharts/plugins/adjustTickAmount.ts]: The param `axes` is only yAxis. should change name of function and parameter to context of yAxis.
e.g. `getBaseYAxis(yaxes: IHighchartsAxisExtend[])`

Comment (naibinh) [reply src/components/visualizations/chart/highcharts/plugins/adjustTickAmount.ts]: done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4314/ : SUCCESS in 8m 31s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5501/ : SUCCESS in 9s

**Pull request closed without merging**